### DISCUSSION TITLE
Boundary treatment for the velocity divergence calculation

### DISCUSSION THREAD

**Discussion by mattfalcone1997** [xcompact3d/Incompact3d#227] [General]
Hi,

I am having a closer look at how the velocity divergence is computed in the boundary cells when  `nclx=2` and how the compact difference stencil is modified.  I am particularly interested in the left-hand side of eq. 20 in Laizet and Lamballais 2009. I assume it uses ghost cells, but I'm unsure how. Line 741 in schemes.f90 in the current master suggests that the diagonal at `nxm` (`ccx6(nxm)`) is $1+\alpha$ with the left of the diagonal (`cbx6(nxm)`) being 0, but I am not sure why. Also to make sure that I am correct, I believe the right-hand side (line 3893 of derive.f90 ) extrapolates the ghost value: $f_{n+2} = f_n +(f_n - f_{n-1})=2f_n - f_{n-1}$. Obviously, I may be looking at the wrong bits of the code altogether.

Best wishes and hopefully my question makes sense!
Matthew

**Comment by mathrack** [comment]
Hello,

I think `ccx6` is used for the staggered derivation, and we have ghost points for this operator.

In the line
```
              tx(nxm,j,k)=acix6*(ux(nx,j,k)-ux(nxm,j,k))&
                         +bcix6*(two*ux(nx,j,k)-ux(nxm,j,k)-ux(nxm-1,j,k))
```
The underlying equation is
```
              tx(nxm,j,k)=acix6*(ux(nx,j,k)-ux(nxm,j,k))&
                         +bcix6*(ux(nx+1,j,k)-ux(nxm-1,j,k))
```
Thus, I think we have $f_{n+1} = f_n + (f_n-f_{n-1}) = 2 f_n - f_{n-1}$
